vm/devices/storage/scsi_buffers/src/lib.rs

```diff
@@ -204,6 +205,52 @@ impl LockedRange for LockedIoVecs {
     }
 }
 
+/// An implementation of [`MemoryWrite`] that provides semantically
+/// correct results. Specifically, it always returns a `ReadOnly` error
+/// when attempting to write to it.
+pub struct PermissionedMemoryWriter<'a> {
+    range: PagedRange<'a>,
+    writer: PagedRangeWriter<'a>,
+    is_write: bool,
+}
+
+impl PermissionedMemoryWriter<'_> {
+    /// Creates a new memory writer with the given range and guest memory.
+    pub fn new<'a>(
```

Comment:
Even if we decided to keep `PermissionedMemoryWriter` `pub` (to make the return value of `writer` explicit), I wouldn't make `new` `pub`--there's no reason for this crate to provide this as a public interface to other crates.

(Perhaps this could be a useful addition to `guestmem::ranges`, so you could move it there and keep it `pub`, if you want to.)

Reply:
Perhaps this is a useful addition, but I'm trying to keep this change lightweight so will keep it here.